Sources/Sentry/SentryScope.m

```diff
@@ -251,142 +319,132 @@ - (void)setLevel:(enum SentryLevel)level
 
 - (void)addAttachment:(SentryAttachment *)attachment
 {
-    @synchronized(self.attachmentArray) {
-        [self.attachmentArray addObject:attachment];
+    @synchronized(_attachmentArray) {
+        [_attachmentArray addObject:attachment];
     }
 }
 
 - (NSArray<SentryAttachment *> *)attachments
 {
-    @synchronized(self.attachmentArray) {
-        return self.attachmentArray.copy;
+    @synchronized(_attachmentArray) {
+        return _attachmentArray.copy;
     }
 }
 
 - (NSDictionary<NSString *, id> *)serialize
 {
-    @synchronized(self) {
-        NSMutableDictionary *serializedData = [NSMutableDictionary new];
-        [serializedData setValue:[self.tagDictionary copy] forKey:@"tags"];
-        [serializedData setValue:[self.extraDictionary copy] forKey:@"extra"];
-        [serializedData setValue:[self.contextDictionary copy] forKey:@"context"];
-        [serializedData setValue:[self.userObject serialize] forKey:@"user"];
-        [serializedData setValue:self.distString forKey:@"dist"];
-        [serializedData setValue:self.environmentString forKey:@"environment"];
-        [serializedData setValue:[self.fingerprintArray copy] forKey:@"fingerprint"];
-        if (self.levelEnum != kSentryLevelNone) {
-            [serializedData setValue:SentryLevelNames[self.levelEnum] forKey:@"level"];
-        }
-        NSArray *crumbs = [self serializeBreadcrumbs];
-        if (crumbs.count > 0) {
-            [serializedData setValue:crumbs forKey:@"breadcrumbs"];
-        }
-        return serializedData;
+    NSMutableDictionary *serializedData = [NSMutableDictionary new];
+    [serializedData setValue:[self tags] forKey:@"tags"];
+    [serializedData setValue:[self extras] forKey:@"extra"];
+    [serializedData setValue:[self context] forKey:@"context"];
+    [serializedData setValue:[self.userObject serialize] forKey:@"user"];
+    [serializedData setValue:self.distString forKey:@"dist"];
+    [serializedData setValue:self.environmentString forKey:@"environment"];
+    [serializedData setValue:[self fingerprints] forKey:@"fingerprint"];
+
+    if (self.levelEnum != kSentryLevelNone) {
```

Comment:
We need a copy of `levelEnum` since setter can be called between reads

Reply:
Good catch 👍🏼